lib/main.dart

```diff
@@ -28,14 +33,15 @@ class SynchronizationService {
 
   Future<void> performSynchronizationTask() async {
     try {
-      await syncBlockchain();
+      await syncBlockchain(network: network);
```

Comment:
I think it's better to request the network block height only after pressing 'create wallet', then we don't have to deal with this 'SynchronizationService'/'PerformSynchronizationTask' stuff. This is related to the previous comment (don't use WalletState to keep track of the network)